lib/dma.c

```diff
@@ -599,11 +600,32 @@ dma_controller_dirty_page_get(dma_controller_t *dma, vfu_dma_addr_t addr,
         return ERROR_INT(EINVAL);
     }
 
-    memcpy(bitmap, region->dirty_bitmap, size);
+    for (i = 0; i < (size_t)bitmap_size; i++) {
```

Comment:
Not required for this patch, but should we consider in the future using QWORD instead of bytes for performance?

Reply:
fair enough